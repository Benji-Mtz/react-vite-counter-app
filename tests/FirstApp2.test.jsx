import { render, screen } from "@testing-library/react";
import { FirstApp } from "../src/FirstApp";

describe('Pruebas en <FirstApp />', () => { 
    
    const title = 'Hola, Soy el titulo';
    const subTitle = 'Soy un subtitulo';
    
    test('debe de hacer match con el snapshot', () => {
               
        const { container } = render( <FirstApp title={ title } /> );
        expect( container ).toMatchSnapshot();

    });

    test('debe de mostrar el mensaje "Hola, Soy el titulo"', () => { 
        // screen.debug(); => <Body />

        render( <FirstApp title={ title } /> );
        // expect( screen.getByText( title ) ).not.toBeTruthy();
        expect( screen.getByText( title ) ).toBeTruthy();
    });

    test('debe de mostrar el titulo en un h1', () => { 
        render( <FirstApp title={ title } /> )
        // getByRole son todas las etiquetas de HTML
        expect( screen.getByRole('heading', { level: 1 }).innerHTML ).toContain( title );
    });

    test('debe mostrar el subtitulo enviado por props', () => { 
        render( 
            <FirstApp 
                title={ title }
                subTitle={ subTitle }
            /> 
        );

        // expect( getAllByText(subTitle).length ).toBe(2);
        expect( screen.getAllByText( subTitle ).length ).toBe(2);

    });

 });