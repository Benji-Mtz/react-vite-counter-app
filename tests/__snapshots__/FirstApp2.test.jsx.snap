// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <FirstApp /> debe de hacer match con el snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
     
    Hola, Soy el titulo
     
  </h1>
  <p>
    No hay subtítulo
  </p>
  <p>
    No hay subtítulo
  </p>
  <p>
    Benji Martinez
  </p>
</div>
`;
